dvc/remote/base.py

```diff
@@ -550,35 +550,33 @@ def download(
 
         makedirs(to_info.parent, exist_ok=True, mode=dir_mode)
 
-        print(from_info)
-
         if self.isdir(from_info):
             makedirs(to_info, exist_ok=True, mode=dir_mode)
-            # TEMP limits to test
-            limit = 5
-            current = 0
-            # / TEMP limits to test
-            for file_from_info in self.walk_files(from_info):
-                # TEMP limits to test
-                current += 1
-                if current == limit:
-                    exit()
-                # / TEMP limits to test
-
-                file_to_info = Path(
-                    str(to_info)
-                    + "/"
-                    + str(file_from_info)[len(str(from_info)) + 1 :]
-                )
+            file_to_infos = (
+                to_info / file_to_info.relative_to(from_info)
+                for file_to_info in self.walk_files(from_info)
+            )
 
-                self.single_file_download(
+            with ThreadPoolExecutor(max_workers=self.JOBS) as executor:
+                file_from_info = list(self.walk_files(from_info))
```

Comment:
Let's move it up to file_to_infos and use it there instead of self.walk_files(), so we don't walk that directory twice.

Reply:
Btw, better rename it too. Maybe `from_infos` and `to_infos`?